src/OrleansGetEventStore/GetEventStoreProvider.cs

```diff
@@ -0,0 +1,234 @@
+                        position += writePageSize;
+                    }
+
+                    await transaction.CommitAsync();
+                }
+                catch (WrongExpectedVersionException)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static List<KeyValuePair<Guid, object>> emptyList = new List<KeyValuePair<Guid, object>>();
+        private const int maxPageSize = 500;
+
+        public override async Task<EventStreamResponse> LoadStream(string streamName, int fromVersion, int? toVersion)
+        {
+            var sliceStart = fromVersion;
+            StreamEventsSlice currentSlice;
+
+            var events = new List<KeyValuePair<Guid,object>>();
+
+            do
+            {
+                var readPageSize = Math.Min(maxPageSize, (toVersion ?? int.MaxValue) - sliceStart);
+
+                if (readPageSize == 0)
+                    break;
+
+                currentSlice = await this.Connection.ReadStreamEventsForwardAsync(streamName, sliceStart, readPageSize, true);
+
+                if (currentSlice.Status == SliceReadStatus.StreamNotFound 
+                    || currentSlice.Status == SliceReadStatus.StreamDeleted)
+
+                    // stream does not exist... report this as an empty stream.
+                    return new EventStreamResponse()
+                    { 
+                        StreamName = streamName,
+                        Version = 0,
+                        Events = emptyList
+                    };
+
+                foreach (var evnt in currentSlice.Events)
+                      events.Add(new KeyValuePair<Guid,object>(evnt.Event.EventId, DeserializeEvent(evnt.Event)));
+
+                sliceStart = currentSlice.NextEventNumber;
+
+            } while (!currentSlice.IsEndOfStream);
+
+            return new EventStreamResponse()
+            {
+                StreamName = streamName,
+                Version = fromVersion + events.Count,
+                Events = events
+            };
+        }
+
+        public static object DeserializeEvent(RecordedEvent @event)
+        {
+            // Backwards compatibility
+            var metadata = DeserializeMetadata(@event.Metadata);
+
+            var eventTypeProperty = metadata.Property("EventClrTypeName");
+
+            if (eventTypeProperty != null)
+                return DeserializeEvent((string)eventTypeProperty.Value, @event.Data);
+            // Backwards compatibility
+
+            return DeserializeEvent(@event.EventType, @event.Data);
+        }
+
+        public static JObject DeserializeMetadata(byte[] metadata)
+        {
+            return JObject.Parse(Encoding.UTF8.GetString(metadata));
+        }
+
+        public static object DeserializeEvent(string eventClrTypeName, byte[] data)
+        {
+            return JsonConvert.DeserializeObject(Encoding.UTF8.GetString(data), Type.GetType(eventClrTypeName));
```

Comment:
It is not about using the static method or not, but it is about configuring settings explicitly.
If the user is not supplying his own settings for ```JsonSerialization``` we've to have a default of or own, since relying on ```NewtonSoft.Json``` default settings  can carry some uncertainty.